Build.cmake

```diff
@@ -101,35 +101,57 @@ ENDIF()
    ENDIF (MSVC OR MINGW)
 
    set(SRC_FILES ${SRC_FILES} ${SRC_PLATFORM_SPECIFIC})
-
-   # Create the g3log library
    INCLUDE_DIRECTORIES(${LOG_SRC})
-  SET(G3LOG_LIBRARY g3logger)
```

Comment:
I would keep this so that the logger name is *always*  the same. 

Do you see an issue with building the library with only one of : static or shared?  If so the test library can always just link to ```${G3LOG_LIBRARY}``` it will just be staic *or* shared depending on your cmake configuraiton option (see comment below)